### ISSUE TITLE
Functionality to replace password in keyring store

### ISSUE THREAD

**Issue by takluyver** [takluyver/flit#113]
If you have `keyring` installed, and no password in your `.pypirc` file, flit will prompt you for your password the first time you upload, and then store it with keyring and reuse it for future uploads. But if you type the password incorrectly, or you change your password on the server, flit will still try to reuse the password it has stored. You would have to use the platform UI, which is often unfamiliar, to find and change/remove the stored password.

**Comment by glyph** [comment]
Is there any irony in the fact that https://github.com/glyph/secretly was the first project I released using flit, a day after you filed this issue? :) (I did not read it at the time.)

**Comment by takluyver** [comment]
Ha! I'm flattered that you're using it :-)

**Comment by melMass** [comment]
Just commenting since I just struggled to find this myself on windows it's in the Credential Manager (`Control Panel\All Control Panel Items\Credential Manager`): 
- Run (Windows + R)
- Type `control /name Microsoft.CredentialManager` and enter, find the row with pypi.org and hit remove

![image](https://github.com/pypa/flit/assets/7041726/fd1bd305-9198-4e7d-a24b-eb770463c4c5)

**Comment by Aran-Fey** [comment]
It's downright criminal that nothing has been done about this in 7 years. The error message you get from an invalid password is exceedingly unhelpful:

> See https://pypi.org/help/#invalid-auth for more information. for url: https://upload.pypi.org/legacy/

It sends you down *two* rabbit holes that have nothing to do with your problem, instead of simply telling you "hey, the password was incorrect, please input the correct one:".

**Comment by glyph** [comment]
> It's downright criminal that nothing has been done about this in 7 years

You're right, it is weird that you haven't submitted a PR yet. Why haven't you?  I couldn't find one with a [quick search](https://github.com/pypa/flit/pulls?q=is%3Apr+is%3Aopen+keyring), did you forget to open one to your branch where you've implemented this?

(To be clear: I am satirizing your post because this is an inappropriately aggressive tone to take with a volunteer project, and you are as much a possible volunteer as Thomas here. If it's important to you, submit a fix and advocate for its inclusion. If you think that this makes Flit inappropriate to use, it's perfectly fair to tell other people not to use it because volunteer projects are not above criticism. But yelling at the maintainers doesn't help anyone.)

**Comment by Aran-Fey** [comment]
> Why haven't you?

Because that is a ridiculously inefficient way of getting things done. I have my own projects to work on, and I wouldn't expect my users so submit a PR for such a trivial change. Do you seriously think that developers should familiarize themselves with every code base that they have feature requests for? If the world worked like that, then every code base on the planet would be an absolute mess because it's mostly written by clueless people.

If you think my tone was too aggressive, that's fair enough. But the "argument" that I should submit a PR is laughable.

**Comment by takluyver** [comment]
@glyph already pointed out that his post was satire to highlight the unreasonable tone you took. Of course you don't have to contribute, but you don't get to lecture other people on what they should spend their time doing. This is not a product, you are not a customer.

**Comment by Aran-Fey** [comment]
Can you point out where I told someone what they should spend their time doing?

I posted two things: facts and my opinion.

**Comment by takluyver** [comment]
The tone of your first post was pretty clear, and this attitude is not welcome. Please stop arguing - it's not going to achieve anything.
